docs/dev_guide/model.md

```diff
@@ -0,0 +1,310 @@
+We are using the `KerasCategorical` model as an example. The loss function here
+makes explicit usage of the output tensors of the
+model (`angle_out, throttle_out`).
+
+```python
+def x_transform(self, record: TubRecord):
+    img_arr = record.image(cached=True)
+    return img_arr
+```
+
+In this function you define how to extract the input data from your
+recorded data. This data is usually called `X` in the ML frame work . We have
+shown the implementation in the base class which works for all models that have
+only the image as input. 
+
+The function returns a single data item if the model has only one input. You 
+need to return a tuple if your model uses more input data.
+**Note:** _If your model has more inputs, the tuple needs to have the image in 
+the first place._ 
+
+```python
+def y_transform(self, record: TubRecord):
+    angle: float = record.underlying['user/angle']
+    throttle: float = record.underlying['user/throttle']
+    return angle, throttle
+```
+In this function you specify how to extract the `y` values (i.e. target
+values) from your recorded data.
+
+
+```python
+def x_translate(self, x: XY) -> Dict[str, Union[float, np.ndarray]]:
+    return {'img_in': x}
+```
+Here we require a translation of how the `X` value that you extracted above will
+be fed into `tf.data`. Note, `tf.data` expects a dictionary if the model has
+more than one input variable, so we have chosen to use dictionaries also in the
+one-argument case for consistency. Above we have shown the implementation in the
+base class which works for all models that have only the image as input. You
+don't have to overwrite neither `x_transform` nor
+`x_translate` if your model only uses the image as input data.
+**Note:** _the keys of the dictionary must match the name of the **input**  
+layers in the model._
+
+```python
+def y_translate(self, y: XY) -> Dict[str, Union[float, np.ndarray]]:
+    if isinstance(y, tuple):
+        angle, throttle = y
+        return {'angle_out': angle, 'throttle_out': throttle}
+    else:
+        raise TypeError('Expected tuple')
+```
+Similar to the above, this provides the translation of the `y` data into the
+dictionary required for `tf.data`. This example shows the implementation of
+`KerasLinear`.
+**Note:** _the keys of the dictionary must match the name of the **output**
+layers in the model._
+
+```python
+def output_shapes(self):
+    # need to cut off None from [None, 120, 160, 3] tensor shape
+    img_shape = self.get_input_shape()[1:]
+    shapes = ({'img_in': tf.TensorShape(img_shape)},
+              {'angle_out': tf.TensorShape([15]),
+               'throttle_out': tf.TensorShape([20])})
+    return shapes
+```
+This function returns a tuple of _two_ dictionaries that tells tensorflow which
+shapes are used in the model. We have shown the example of the 
+`KerasCategorical` model here.
+**Note 1:** _The keys of the two dictionaries must match the name of the
```

Comment:
Nit: Call this **Note** and not **Note 1**. 
Add a paragraph like the ones above.

Reply:
I have 2 notes there...